### PR TITLE
Introduce the Authentication package

### PR DESCRIPTION
**Pull request by mvriel**
In this package, I have introduced the concept of a Session. A Session represents the means to sign in into a service, such as Azure AD. With a session you do not need to be signed -i.e. anonymous-.

The Session is a ScriptableObject so that it can be re-used across scenes; and an Authenticator component is introduced to listen to the transitions that can happen and handle signing in using the given session.

This package relies on the CDMVision's Authentication package; which provides a good OAuth client that we wrap to make it easier to use; and to introduce an Azure AD version of the OAuth client. For WebGL we needed to find another way because it doesn't support WebGL; for that I make use of the JSO oAuth library and a jslib to get the accesstoken.

To conveniently make requests to webservices that need the accesstoken, there is a SignWebRequest method in the Session object that can manage the signing of a webrequest with a suitable token.

### CODE REVIEW

**Comment by mvriel** [comment]
The package needs a sample scene, I need to properly test it in WebGL and the package needs to include all necessary artefacts to conform to Unity's package standards. Including a README.md with usage instructions

**Comment by mvriel** [comment]
WebGL issues are caused by the use of, at least, HttpClient. I have opened an issue to get input from the package author: https://github.com/cdmvision/authentication-unity/issues/11

File: Packages/manifest.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "dependencies": {
+    "com.cdm.authentication": "https://github.com/cdmvision/authentication-unity.git#1.1.3",
```
Comment (mvriel) [inline Packages/manifest.json:3]: I have opened an issue at https://github.com/cdmvision/authentication-unity/issues/13 to add this library to OpenUPM; when that is done, we can add it as a dependency to the package itself

File: README.md [hunk 2]
```diff
@@ -30,10 +30,21 @@ If you imported a Netherlands3D package and like to update the package, or pick
...
-## Hosting a WebGL Netherlands3D application
+## Running a WebGL Netherlands3D application
 
 If your project uses the TileHandler, and you would like to use the compressed binary tile files (ending in .br) in a WebGL build, you need to make sure your host server (remote or localhost) has the 'Content-Encoding' header set to 'br' for all files that have the .br file extention.
 
+### Local
+
+To develop and test the WebGL locally, it is recommended to build your WebGL output into the Build folder. You can 
+either use the "Build and Run" option, or you can start a local webserver that runs the application on 
+http://localhost:8080 by running `./bin/start-server.bat` or `./bin/start-server.sh`.
```
Comment (sambaas) [inline README.md:41]: handig! misschien wel goed hierbij te vermelden dat dit afhankelijk is van een python installatie?

File: Packages/nl.netherlands3d.authentication/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "licensesUrl": "https://github.com/Amsterdam/Netherlands3D/blob/main/Packages/nl.netherlands3d.authentication/LICENSE.txt",
+  "changelogURL": "https://github.com/Amsterdam/Netherlands3D/blob/main/Packages/nl.netherlands3d.authentication/CHANGELOG.md",
+  "hideInEditor": false,
+  "author": {
+    "name": "3D Amsterdam Team",
+    "url": "https://3d.amsterdam.nl"
+  },
+  "repository": {
+    "url": "https://github.com/Amsterdam/Netherlands3D"
+  }
```
Comment (sambaas) [inline Packages/nl.netherlands3d.authentication/package.json]: Hier ontbreekt nog de verwijzing naar de sample scene waardoor ik hem niet kan importeren vanaf de package manager.

"samples": [
    {
      "displayName": "Sample name",
      "description": "Sample description",
      "path": "Samples~/"
    },

**Comment by mvriel** [comment]
As discussed, Azure AD requires PKCE; and this is not supported by our current JS library; I should use https://www.jsdelivr.com/package/npm/@badgateway/oauth2-client and then we can try Azure again

**Review by sambaas — APPROVED** [review]
ziet er netjes en begrijpelijk uit allemaal, chapeau 👌

File: Packages/nl.netherlands3d.authentication/package.json (hunk 3 above)
Comment (mvriel) [reply Packages/nl.netherlands3d.authentication/package.json]: Dit is opgelost

**Pull request merged**